lib/chartist-plugin-legend.js

```diff
@@ -176,25 +196,10 @@
                         }
                     }
 
-                    var newSeries = [];
-                    var newLabels = [];
-
-                    for (var i = 0; i < seriesMetadata.length; i++) {
-                        if (seriesMetadata[i].legend != -1 && legends[seriesMetadata[i].legend].active) {
-                            newSeries.push(seriesMetadata[i].data);
-                            newLabels.push(seriesMetadata[i].label);
-                        }
-                    }
-
-                    chart.data.series = newSeries;
-                    if (useLabels) {
-                        chart.data.labels = newLabels;
-                    }
-
-                    chart.update();
+                    applyLegendStateToChart(legends, seriesMetadata, useLabels);
 
                     if (options.onClick) {
-                        options.onClick(chart, e);
+                        options.onClick(chart, e, legends);
```

Comment:
A little odd to pass the internal state out by-reference - since the intent isn't to allow mutation this way.  Maybe pass `_.clone(legends)`

Reply:
Or `_.cloneDeep`